### PR TITLE
Feature/jwt

### PR DESCRIPTION
**Pull request by junbeomlee**
## Related Issue

resolve: #68 #11 

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] Test case
- [x] End of Work

### CODE REVIEW

File: server/src/app/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+export class AuthService {
+    constructor(
+        @Inject('JwtService') private readonly jwtService: JwtService,
+        @Inject('UserService') private readonly userService: UserServiceImpl,
+        @Inject('ValidationService') private readonly validationService: ValidationServiceImpl,
+    ) {}
+
+    createToken(address: string): Token {
+        if (address === undefined) {
+            throw new Error('address should be defined.');
```
Comment (ttkmw) [inline server/src/app/auth/auth.service.ts]: i think should throw invalidparameter error

File: server/src/app/auth/auth.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+      service = module.get<AuthService>(AuthService);
+      expect(service).toBeDefined();
+    });
+  });
+
+  describe('#createToken()', () => {
+    it('should return a valid token', async () => {
+      const result = {
+        expiresIn: 3600,
```
Comment (ttkmw) [inline server/src/app/auth/auth.service.spec.ts]: i think extracting expiresIn and accessToken is better

File: server/src/app/auth/auth.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(false);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
```
Comment (ttkmw) [inline server/src/app/auth/auth.service.spec.ts]: i thinkk toThrow(`Error`) is needed in all the spec files

File: server/src/app/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    async signIn(address: string, message: string, signature: string): Promise<SignInResponseDto> {
+        const result: boolean = this.validationService.verify(address, message, signature);
+        if (!result) {
+            throw new UnauthorizedException();
+        }
+
+        const user: User = await this.userService.getByAddress(address);
+        if (user === undefined) {
+            throw new Error('user not found');
```
Comment (ttkmw) [inline server/src/app/auth/auth.service.ts]: i think throw notfoundexception is better

File: server/src/app/auth/jwt.strategy.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export class JwtStrategy extends PassportStrategy(Strategy) {
+    constructor(private readonly authService: AuthService) {
+        super({
+            jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+            secretOrKey: 'secretKey',
+        });
+    }
+
+    async validateToken(address: string) {
+        const user = await this.authService.createToken(address);
```
Comment (ttkmw) [inline server/src/app/auth/jwt.strategy.ts]: i think 'checking address is stored in database ' is needed
isn't it all the tokens are validated if address is exists in dto?

and im just curious, when the address is injected?

File: server/src/domain/user/validation.service.impl.spec.ts [hunk 6]
```diff
@@ -5,6 +5,7 @@ import {BridgeService} from './bridge.service';
 import {Web3Module} from '../../port/adapter/service/blockchain/ethereum/web3/web3.module';
 import {ConfigModule} from 'nestjs-config';
 import * as path from 'path';
+import { Web3BridgeService } from '../../port/adapter/service/blockchain/ethereum/web3/web3.bridge.service';
```
Comment (ttkmw) [inline server/src/domain/user/validation.service.impl.spec.ts]: is it ok to inject 'port' to 'domain'? if it is ok, removing
```
class BrideServiceImpl implements BridgeService {
        recover(message: string, signature: string): string {
            return '';
        }
    }
```
is needed

File: server/src/app/auth/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        @Inject('ValidationService') private readonly validationService: ValidationServiceImpl,
+    ) {}
+
+    createToken(address: string): Token {
+        if (address === undefined) {
+            throw new Error('address should be defined.');
+        }
+
+        return {
+            expiresIn: 3600,
```
Comment (ttkmw) [inline server/src/app/auth/auth.service.ts:24]: i think set config using config file is better

is it next feature?

**Review by ttkmw — COMMENTED** [review]
LGTM except comment!

File: server/src/app/auth/auth.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+        .thenReturn(undefined);
+
+      when(mockJwtService.sign(address)).thenReturn(undefined);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
```
Comment (inDlife) [inline server/src/app/auth/auth.service.spec.ts]: i thinkk toThrow(Error) is needed

File: server/src/app/auth/auth.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+
+      when(mockValidationService.verify(address, message, signature))
+        .thenReturn(false);
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
```
Comment (inDlife) [inline server/src/app/auth/auth.service.spec.ts]: i thinkk toThrow(Error) is needed

File: server/src/app/auth/auth.service.spec.ts [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+        .thenReturn(new Promise((resolve) => {
+          resolve(undefined);
+        }));
+
+      service = new AuthService(instance(mockJwtService),
+        instance(mockUserService),
+        instance(mockValidationService));
+
+      await expect(service.signUp(address, name, message, signature))
+        .rejects.toThrow();
```
Comment (inDlife) [inline server/src/app/auth/auth.service.spec.ts]: i thinkk toThrow(Error) is needed

File: server/src/app/auth/auth.service.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+                address,
+                expiresIn: 3600,
+            }),
+        };
+    }
+
+    async signUp(address: string, name: string, message: string, signature: string): Promise<SignUpResponseDto> {
+        const result: boolean = this.validationService.verify(address, message, signature);
+        if (!result) {
+            throw new UnauthorizedException();
```
Comment (ttkmw) [inline server/src/app/auth/auth.service.ts]: i think giving message is better( in `siginIn` too)

File: server/src/domain/user/validation.service.impl.ts [hunk 12]
```diff
@@ -1,13 +1,14 @@
 import {Inject, Injectable} from '@nestjs/common';
-import {BridgeService} from './bridge.service';
 import {ValidationService} from './validation.service';
+import { BridgeService } from './bridge.service';
+import { Web3BridgeService } from '../../port/adapter/service/blockchain/ethereum/web3/web3.bridge.service';
```
Comment (ttkmw) [inline server/src/domain/user/validation.service.impl.ts]: i think it should be deleted

File: server/src/port/module/auth.module.ts [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+@Module({
+    imports: [
+        PassportModule.register({ defaultStrategy: 'jwt' }),
+        JwtModule.register({
+                secretOrKeyProvider: () => {
+                    return 'secretKey';
+                },
+            signOptions: {
+                expiresIn: '10h',
```
Comment (ttkmw) [inline server/src/port/module/auth.module.ts]: access token expires in 3600.
why 10h is here?
is it different from access token?

File: server/src/web/auth.controller.spec.ts [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+      when(mockAuthService.signUp(anyString(), anyString(), anyString(), anyString())).thenReturn(new Promise(resolve => {
+        resolve(authDto);
+      }));
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.signUp(signUpRequestDto)).toBe(authDto);
+    });
+
+    it('should return null', async () => {
```
Comment (ttkmw) [inline server/src/web/auth.controller.spec.ts]: i think should retrun undefined is right

File: server/src/web/auth.controller.spec.ts [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+      when(mockAuthService.signIn(address, message, signature)).thenReturn(new Promise(resolve => {
+        resolve(authDto);
+      }));
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.signIn(signInRequestDto)).toBe(authDto);
+    });
+
+    it('should throw "user not found"', async () => {
```
Comment (ttkmw) [inline server/src/web/auth.controller.spec.ts]: it says should throw error but returns undefined at the code

File: server/src/web/auth.controller.spec.ts [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+      when(mockAuthService.signUp(address, name, message, signature))
+        .thenReject().thenThrow(new UnauthorizedException());
+
+      controller = new AuthController(instance(mockAuthService));
+
+      expect(await controller.createToken(createTokenRequestDto))
```
Comment (ttkmw) [inline server/src/web/auth.controller.spec.ts]: can not understand why testing `createToken` at the `signIn` test

File: server/src/app/auth/auth.service.ts (hunk 7 above)
Comment (sojeongw) [reply server/src/app/auth/auth.service.ts:24]: @inDlife After discussing it with other reviewers, I removed the config file and changed it to hard-coded style.

**Pull request closed without merging**